libs/blocks/quiz/quiz.js

```diff
@@ -0,0 +1,338 @@
+  const [btnAnalytics, setBtnAnalytics] = useState(null);
+  const [btnClicked, setBtnClicked] = useState(false);
+  const initialUrlParams = getUrlParams();
+  const [urlParam, setUrlParam] = useState(initialUrlParams);
+  const VALID_QUESTIONS = useMemo(() => [], []);
+  const KNOWN_PARAMS = useMemo(() => ['martech', 'milolibs', 'quiz-data'], []);
+
+  useEffect(() => {
+    (async () => {
+      const [questions, dataStrings] = await getQuizData();
+      const qMap = {};
+      questions.questions.data.forEach((question) => {
+        qMap[question.questions] = question;
+        VALID_QUESTIONS.push(question.questions);
+      });
+
+      const strMap = {};
+      dataStrings.questions.data.forEach((question) => {
+        strMap[question.q] = question;
+      });
+
+      setUserFlow([questions.questions.data[0].questions]);
+
+      setStringData(dataStrings);
+      setQuestionData(questions);
+      setStringQList(strMap);
+      setQuestionList(qMap);
+
+      // wait for data to load
+      setDataLoaded(true);
+
+      // add quiz class to page
+      document.body.classList.add('quiz-page');
+    })();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [setQuestionData, setStringData, setStringQList, setQuestionList]);
+
+  useEffect(() => {
+    function handlePopState() {
+      const currentURL = window.location.href;
+      window.location.href = currentURL;
+    }
+    if (isDataLoaded) {
+      window.addEventListener('popstate', handlePopState);
+      return () => {
+        window.removeEventListener('popstate', handlePopState);
+      };
+    }
+    return () => {};
+  }, [KNOWN_PARAMS, isDataLoaded]);
+
+  useEffect(() => {
+    if (userFlow.length) {
+      const currentFlow = userFlow.shift();
+      if (!currentFlow?.length) {
+        return;
+      }
+      setSelectedQuestion(questionList[currentFlow] || []);
+    }
+  }, [userFlow, questionList]);
+
+  /**
+   * Updates the analytics data for the next button.
+   * Happens with each option click/tap.
+   * @returns {void}
+   */
+  useEffect(() => {
+    setBtnAnalytics(getAnalyticsDataForBtn(selectedQuestion, selectedCards));
+  }, [selectedQuestion, selectedCards]);
+
+  /**
+   * Handling the result flow when user has selected all the options.
+   * nextQuizViewsExist is set to false when the next button is clicked
+   * and there are no more views to show.
+   */
+  useEffect(() => {
+    if (!nextQuizViewsExist && userSelection.length) {
+      handleResultFlow(transformToFlowData(userSelection));
+    }
+  }, [userSelection, nextQuizViewsExist]);
+
+  /**
+   *  Updates the url param when user selects the options.
+   *  Happens with each option click/tap.
+   */
+  useEffect(() => {
+    const cardValues = Object.getOwnPropertyNames(selectedCards);
+    if (selectedQuestion) {
+      setUrlParam((prevUrlParam) => {
+        const newParam = { ...prevUrlParam };
+        if (selectedQuestion && cardValues.length === 0) {
```

Comment:
May want to assign `selectedQuestion.questions` to a const with shorter name (`questions`?)